src/ray/raylet/node_manager.h

```diff
@@ -95,6 +95,15 @@ class NodeManager : public rpc::NodeManagerServiceHandler,
   /// Get the port of the node manager rpc server.
   int GetServerPort() const { return node_manager_server_.GetPort(); }
 
+  /// Preprocess request from raylet client. This takes care of common processing for each
```

Comment:
Be specific about each of these steps. Logging is self-explanatory, but for instance, I cannot tell from this comment what "checking heartbeat" means, or what will happen if the "worker is being killed".